src/utils/componentSwapData.ts

```diff
@@ -185,22 +139,19 @@ function getStaticRedemptionSwapData(
   }
 }
 
-export function getUnderlyingErc20(token: string): string {
-  switch (token) {
-    // wfDAI:1695168000 (Wrapped fDAI @ 1695168000)
-    case '0x278039398A5eb29b6c2FB43789a38A84C6085266':
-    // wfDAI:1687392000 (Wrapped fDAI @ 1687392000)
-    case '0xfa5d4F65a4c51906652d78140C266423111c6BFA':
+export async function getUnderlyingErc20(
+  token: string,
+  provider: JsonRpcProvider
+): Promise<string> {
+  const IERC4262_ABI = ['function asset() public view returns (address)']
+  const contract = new Contract(token, IERC4262_ABI, provider)
+  const underlyingERC20: string = await contract.asset()
+  switch (underlyingERC20.toLowerCase()) {
+    case dai.toLowerCase():
       return dai
-    // mcUSDC (Morpho-Compound USD Coin Supply Vault)
-    case '0xba9E3b3b684719F80657af1A19DEbc3C772494a0':
-    // wfUSDC:1695168000 (Wrapped fUSDC @ 1695168000)
-    case '0xe09B1968851478f20a43959d8a212051367dF01A':
+    case usdc.toLowerCase():
       return usdc
-    // maUSDT (Morpho-Aave Tether USD Supply Vault )
-    case '0xAFe7131a57E44f832cb2dE78ade38CaD644aaC2f':
-    // mcUSDT (Morpho-Compound Tether USD Supply Vault)
-    case '0xC2A4fBA93d4120d304c94E4fd986e0f9D213eD8A':
+    case usdt.toLowerCase():
```

Comment:
Changed this to be more dynamic because the components of MMI might change over time.

If there would be cTokens at some point, we'd probably have to update this function - as their the read function is called `underlying()`instead of the IERC4262's `asset()`